src/test_workflow/test_recorder/test_recorder.py

```diff
@@ -46,17 +54,45 @@ def _generate_std_files(self, stdout: str, stderr: str, output_path: str) -> Non
             stderr_file.write(stderr)
 
     def _generate_yml(self, test_result_data: TestResultData, output_path: str) -> str:
+        base_file_path = self._get_file_path(self.base_path, test_result_data.component_name, test_result_data.component_test_config)
+
+        components_files = self._get_list_files(output_path)
+        components_files = self._update_absolute_file_paths(components_files, base_file_path, "")
+
+        local_cluster_logs_path = os.path.join(output_path, "local-cluster-logs")
+        local_cluster_logs = self._get_list_files(local_cluster_logs_path)
+        local_cluster_logs = self._update_absolute_file_paths(local_cluster_logs, base_file_path, "local-cluster-logs")
+
+        opensearch_service_logs_path = os.path.join(local_cluster_logs_path, "opensearch-service-logs")
+        opensearch_service_logs = self._get_list_files(opensearch_service_logs_path)
+        opensearch_service_logs = self._update_absolute_file_paths(opensearch_service_logs, base_file_path, os.path.join("local-cluster-logs", "opensearch-service-logs"))
+
+        test_result_file = components_files + [
+            {
+                "local-cluster-logs":
+                    local_cluster_logs + opensearch_service_logs
+            }
+        ]
```

Comment:
We want this to be generic so that any log files can be written under here. Did you check how this behavior affects dashboards integ-test? Since this just seems to be related to opensearch logs and test_recorder is generic to all tests plus OS and OSD. Should this be part of  `LocalClusterLogs` class? 
https://github.com/opensearch-project/opensearch-build/pull/3497/files#diff-d8e45f5fe0a4500dc4bb059941f629bcc07c6cff714a4a6ee1711dc1fdc7c08aR104

Reply:
@gaiksaya 
For integ test on dashboards, the behavior here are the same. The component yml file (`functionalTestDashboards.yml`) would list out the `stdout&stderr` for component and local cluster logs since it doesn't contain `opensearch_service`. The only things it wouldn't include are those cypress folders (e.g. cypress-screenshots/cypress-videos/cypress-report). I don't think it necessary to keep track of them in the scope of this PR as they are relatively too much to list them all. We could anyway track those videos/reports from the OSD team website I believe. 

I don't think we should have this be part of the `LocalClusterLogs` as that class is mainly for recording and generating those files. Here we basically just list them out in the component yml file.